File: app/src/main/java/com/wilinz/yuetingmusic/ui/settings/SettingsFragment.java
package com.wilinz.yuetingmusic.ui.settings;

import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.View;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.preference.Preference;
import androidx.preference.PreferenceFragmentCompat;

import com.wilinz.yuetingmusic.R;
import com.wilinz.yuetingmusic.util.IntentUtil;

public class SettingsFragment extends PreferenceFragmentCompat {

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        setPreferencesFromResource(R.xml.root_preferences, rootKey);
        try {
            String versionName = requireContext().getPackageManager().getPackageInfo(requireContext().getPackageName(), 0).versionName;
            getPreferenceManager().findPreference("app_version").setSummary(versionName);
        } catch (PackageManager.NameNotFoundException e) {
            e.printStackTrace();
        }

    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        view.setFitsSystemWindows(true);

    }

    @Override
    public boolean onPreferenceTreeClick(@NonNull Preference preference) {
        String key = preference.getKey();
        if (key.equals("app_author")) {
            IntentUtil.browse(requireContext(), preference.getSummary() + "");
        } else if (key.equals("source_code")) {
            IntentUtil.browse(requireContext(), preference.getSummary() + "");
        }
        return super.onPreferenceTreeClick(preference);
    }


}

File: app/src/main/java/com/wilinz/yuetingmusic/ui/settings/SettingsFragment.kt
package com.wilinz.yuetingmusic.ui.settings

import android.content.pm.PackageManager
import android.os.Bundle
import android.view.View
import androidx.preference.Preference
import androidx.preference.PreferenceFragmentCompat
import com.wilinz.yuetingmusic.R
import com.wilinz.yuetingmusic.util.IntentUtil.browse

class SettingsFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        setPreferencesFromResource(R.xml.root_preferences, rootKey)
        try {
            val versionName = requireContext().packageManager.getPackageInfo(
                requireContext().packageName,
                0
            ).versionName
            preferenceManager.findPreference<Preference>("app_version")!!.summary = versionName
        } catch (e: PackageManager.NameNotFoundException) {
            e.printStackTrace()
        }
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        view.fitsSystemWindows = true
    }

    override fun onPreferenceTreeClick(preference: Preference): Boolean {
        val key = preference.key
        if (key == "app_author") {
            browse(requireContext(), preference.summary.toString() + "")
        } else if (key == "source_code") {
            browse(requireContext(), preference.summary.toString() + "")
        }
        return super.onPreferenceTreeClick(preference)
    }
}